lib/model/query/datasets.js

```diff
@@ -122,6 +122,29 @@ const createOrMerge = (parsedDataset, form, fields) => async ({ one, Actees, Dat
   return partial.with({ action: result.action, fields: dsPropertyFields });
 };
 
+// Insert a Dataset when there is no associated form and immediately publish
+const createPublishedDataset = (dataset, project) => async ({ one, Actees }) => {
+  const actee = await Actees.provision('dataset', project);
+  const unpublished = await one(insert(dataset.with({ acteeId: actee.id })))
+    .then((d) => new Dataset(d));
+  return one(sql`${markPublished(unpublished)} returning *`)
```

Comment:
I feel like I'm going overboard on using the generic `insert` (because of the recent bulk entity creation) and then trying to add in a generic `publish` function.  Maybe I should write custom SQL for dataset (and property) insert so I can handle this `publishedAt` timestamp.

Reply:
Yeah, I feel like a generic publish function might be tricky. For `datasets`, we want `publishedAt` to match `createdAt`, but for `ds_properties`, there's a `publishedAt` column but no `createdAt` — so `publishedAt` there should be more like `now()` or `clock_timestamp()`. Maybe you could keep `insert()` but use a short custom `UPDATE` to set `publishedAt`. Or maybe a custom `INSERT` wouldn't be too bad.